### ISSUE TITLE
Suggest: Workbookinfo, Sheetidentifier and publishing state

### ISSUE THREAD

**Issue by came** [tableau/extensions-api#151]
Hi Tableau Team,

I think that dashboard extension is an awesome concept with many valuable use-cases to come for Tableau customers (we are already using the [Tableau JS API](https://onlinehelp.tableau.com/current/api/js_api/en-us/JavaScriptAPI/js_api_ref.htm)).
Looking a little bit into the API I am wondering that a few (I think) essential information are missing (compared to the JS API) when using tableau.extension.*

Especially I haven't found any reference to the containing workbook of the dashboard? There are no unique identifiers for the sheet (dashboard) as well as worksheet? Besides it would be valuable to know if the workbook has already been published or not. 
In summary:
- no workbook references (not even the name?)
- no uniq. identifier of the dashboard (or included sheets) e.g. sheetUrl or viewIds?
- is workbook published / unpublished

Is that because of technical limitations or just because of early phase?
Would be great if you share your view on that.

BR, Carsten

**Comment by bcantoni** [comment]
Closing out based on age. If this is still an issue, please reopen with a comment on the latest status and we can revisit.

**Comment by ttonelli-via** [comment]
@bcantoni Can this issue please be reopened? This functionality still does not exist in the extension api and is something my team is sorely missing

**Comment by bcantoni** [comment]
@ttonelli-via sure, I'll reopen it. Since the earlier description is 4+ years old at this point, could you summarize what your need/request would be? That would be helpful thanks.

**Comment by ttonelli-via** [comment]
@bcantoni Thank you for the quick reply! Specifically for my team's use-case, we are attempting building an extension + server that interfaces with the tableau REST api. The only way to query the REST api for a single dashboard is by an id that is not visible to the extension. The closest thing we are currently able to do is query for all dashboards and filter by dashboard name, but if that name appears in multiple workbooks, there is no information visible to the extension api that will let us deterministically narrow our query down to one dashboard. If the workbook name, path of the dashboard, workbook id, or dashboard id were visible to the extension api, it would solve our issue.

**Comment by Marcelo-Has** [comment]
Hi @bcantoni I'm facing a similar issue.

I need to link a piece of information to a specific dashboard, so when I publish to Tableau Server it my extension search for this dashboard on the database and show the information.

I can't use URL Path because I can't guarantee the developer will be connected and the project will be published when at the time. Also, I can't use the dashboard name for the same @ttonelli-via reason, and the DashboardObjectID is not reliable.

I tested two options. @ttonelli-via one of them may work for you:
• Use a tableau parameter and the extension can write an id (generated by extension) on it. Based on the workbook + dashboard name I can have a unique ID for the dashboard. The problem is you must have a parameter with a specific name all the time, and it's mutable so a developer can erase it by mistake.
• Use settings to save all the information on the workbook. The problem is that if the user removes the extension on the dashboard all information will be lost.

@bcantoni do you have any suggestions? It would be great if Dashboard Extension API could retrieve the Dashboard UUID.
